tesseract_rviz/src/render_tools/env/robot_link.cpp

```diff
@@ -1328,6 +1330,501 @@ bool RobotLink::createEntityForGeometryElement(const shapes::Shape& geom,
+            object->begin("BaseWhiteNoLighting", Ogre::RenderOperation::OT_TRIANGLE_LIST);
+
+            unsigned int vertexCount = 0;
+            Ogre::Vector3 normal(0.0, 0.0, 0.0);
+            for (unsigned int i = 0; i < mesh.getTriangleCount(); ++i)
+            {
+              if (vertexCount >= 2004)
+              {
+                // Subdivide large meshes into submeshes with at most 2004
+                // vertices to prevent problems on some graphics cards.
+                object->end();
+                object->begin("BaseWhiteNoLighting", Ogre::RenderOperation::OT_TRIANGLE_LIST);
+                vertexCount = 0;
+              }
+
+              std::vector<Ogre::Vector3> v(3);
+              std::vector<Ogre::Vector3> normals(3);
+              assert(triangles[(4 * i)] == 3);
+              for (size_t k = 0; k < 3; ++k)
+              {
+                v[k] = Ogre::Vector3(static_cast<float>(vertices[triangles[(4 * i) + (k + 1)]](0)),
+                                     static_cast<float>(vertices[triangles[(4 * i) + (k + 1)]](1)),
+                                     static_cast<float>(vertices[triangles[(4 * i) + (k + 1)]](2)));
+              }
+
+
+              Ogre::Vector3 side1 = v[0] - v[1];
+              Ogre::Vector3 side2 = v[1] - v[2];
+              normal = side1.crossProduct(side2);
+              normal.normalise();
+
+              normals[0] = normal;
+              normals[1] = normal;
+              normals[2] = normal;
+
+              //            float u, v;
+              //            u = v = 0.0f;
+              object->position(v[0]);
+              object->normal(normals[0]);
+              //          calculateUV( verticies[0], u, v );
+              //          object->textureCoord( u, v );
+
+              object->position(v[1]);
+              object->normal(normals[1]);
+              //          calculateUV( verticies[1], u, v );
+              //          object->textureCoord( u, v );
+
+              object->position(v[2]);
+              object->normal(normals[2]);
+              //          calculateUV( verticies[2], u, v );
+              //          object->textureCoord( u, v );
+
+              object->triangle(vertexCount + 0, vertexCount + 1, vertexCount + 2);
+
+              vertexCount += 3;
+            }
+
+            object->end();
+
+            std::string mesh_name = entity_name + "mesh";
+            Ogre::MeshPtr ogre_mesh =
+                object->convertToMesh(mesh_name, Ogre::ResourceGroupManager::DEFAULT_RESOURCE_GROUP_NAME);
+            ogre_mesh->buildEdgeList();
+
+            entity = scene_manager_->createEntity(entity_name, mesh_name);
+
+            delete object;
+          }
+        }
+        break;
+      }
+      case tesseract::CollisionShapeType::CONVEX_MESH:
+      {
+        bool use_resource = false;
+        if (use_resource)
+        {
+          //      const urdf::Mesh& mesh = static_cast<const urdf::Mesh&>(geom);
```

Comment:
Should be commented?

Reply:
Yea, it was originally commented out. It needs to be added to the TODO list.